src/views/docs-view/server.ts

```diff
@@ -345,11 +345,30 @@ export function getStaticGenerationFunctions<
 				layoutProps.githubFileUrl = githubFileUrl
 			}
 
+			/**
+			 * TODO: this logic is temporary to showcase how layouts can work.
+			 * Going forward this would be defined in frontmatter,
+			 * we'd pass frontmatter.layout directly rather than constructing this.
+			 *
+			 * Task: https://app.asana.com/0/1202097197789424/1202850056121889/f
+			 */
+			const isLanding = pathParts.length === 0
+			const customLayout = isLanding
+				? {
+						name: 'docs-root-landing',
+						subtitle:
+							currentRootDocsPath.description ??
+							frontMatter.description ??
+							null,
+				  }
+				: null
+
 			const finalProps = {
 				layoutProps,
 				metadata: {
 					title: frontMatter.page_title ?? null,
 					description: frontMatter.description ?? null,
+					layout: customLayout,
```

Comment:
💭 I think this file is one of our biggest ones in the project, what do you think about abstracting out this logic?

```ts
layout: getCustomLayout(...)
```

Reply:
💯 nice, I dig it - ✅ updated!